src/Sarif.Sarifer/GenerateTestDataCommand.cs

```diff
@@ -37,38 +35,15 @@ public GenerateTestDataCommand(IVsShell vsShell, IMenuCommandService menuCommand
         /// </summary>
         private void MenuCommandCallback(object caller, EventArgs args)
         {
-            this.SendDataToViewerAsync().FileAndForget(FileAndForget.EventName(SendDataToViewerFailureEventName));
-        }
-
-        private async Task SendDataToViewerAsync()
-        {
-            string testDataFilePath = await CreateTestDataFileAsync().ConfigureAwait(continueOnCapturedContext: true);
+            Stream testDataStream = Assembly.GetExecutingAssembly().GetManifestResourceStream(ProofOfConceptResourceName);
 
             // TODO: Why does this never return true?
             if (!viewerInterop.IsViewerExtensionLoaded)
             {
                 this.viewerInterop.LoadViewerExtension();
             }
 
-            await this.viewerInterop.OpenSarifLogAsync(testDataFilePath).ConfigureAwait(continueOnCapturedContext: false);
-        }
-
-        private static async Task<string> CreateTestDataFileAsync()
-        {
-            using (Stream testDataResourceStream = Assembly.GetExecutingAssembly().GetManifestResourceStream("TestData.ProofOfConcept.sarif"))
-            using (TextReader reader = new StreamReader(testDataResourceStream))
-            {
-                // No need to continue on the UI thread because we're just doing file I/O.
-                string testDataFileContents = await reader.ReadToEndAsync().ConfigureAwait(continueOnCapturedContext: false);
-
-                string testDataFilePath = Path.GetTempFileName();
-                using (var writer = new StreamWriter(testDataFilePath))
-                {
-                    await writer.WriteAsync(testDataFileContents).ConfigureAwait(continueOnCapturedContext: false);
-                }
-
-                return testDataFilePath;
-            }
+            this.viewerInterop.OpenSarifLogAsync(testDataStream).FileAndForget(FileAndForget.EventName(SendDataToViewerFailureEventName));
```

Comment:
>testDataStream [](start = 49, length = 14)

Nice!

Reply:
Yeah, right? :-)

---
In reply to: [519462799](https://github.com/microsoft/sarif-visualstudio-extension/pull/259#discussion_r519462799) [](ancestors = 519462799)